# HG changeset patch
# User Sergey Mironov <grrwlf@gmail.com>
# Date 1393407827 0
#      Wed Feb 26 09:43:47 2014 +0000
# Node ID f9ca41851c8831619d50a3434c3466b089aee60d
# Parent  a147ae7b4293171c7bd4b2877ef22fae552a138c
Protect uw_Basis_new_client_source from invalid ctx->id

We assume that FFI code may create new contextes with id left unassigned

diff --git a/src/c/urweb.c b/src/c/urweb.c
--- a/src/c/urweb.c
+++ b/src/c/urweb.c
@@ -1586,6 +1586,9 @@
   int len;
   size_t s_len = strlen(s);
 
+  if(ctx->id < 0)
+    uw_error(ctx, FATAL, "Attempt to create client source using inappropriate context");
+
   uw_check_script(ctx, 15 + 2 * INTS_MAX + s_len);
   sprintf(ctx->script.front, "s%d_%llu=sc(exec(%n", ctx->id, ctx->source_count, &len);
   ctx->script.front += len;
